docs/src/main/sphinx/develop/connectors.rst

```diff
@@ -238,17 +307,21 @@ the strategy employed by the Example HTTP connector.
 .. _connector-record-set-provider:
 
 ConnectorRecordSetProvider
-^^^^^^^^^^^^^^^^^^^^^^^^^^
+--------------------------
 
-Given a split and a list of columns, the record set provider is
+Given a split, a table handle, and a list of columns, the record set provider is
 responsible for delivering data to the Trino execution engine.
-It creates a ``RecordSet``, which in turn creates a ``RecordCursor``
-that's used by Trino to read the column values for each row.
+
+It creates a ``RecordSet``, which in turn creates a ``RecordCursor`` that's used
+by Trino to read the column values for each row. It can use any additional
+information stored in the table handle object by methods the metadata service
```

Comment:
The information can be stored in the table handle or elsewhere. Conceptually, a "table handle" is just an opaque handle to a virtual table. It could be implemented as an id into a remote distributed store that keeps the details, or kept inline inside the handle itself.

It's better to think of the table handle as generalized identifier or key.

Reply:
Thanks! I reworded it.